### PR TITLE
Introduce skeletons for various stubs.

### PR DESCRIPTION
**Pull request by cschuet**
Adds skeletons for

* MapBuilderStub
* PoseGraphStub
* TrajectoryBuilderStub

[RFC=0002](https://github.com/googlecartographer/rfcs/blob/master/text/0002-cloud-based-mapping-1.md)

### CODE REVIEW

File: cartographer_grpc/mapping/pose_graph_stub.h [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  PoseGraphStub(const PoseGraphStub&) = delete;
+  PoseGraphStub& operator=(const PoseGraphStub&) = delete;
+
+  void RunFinalOptimization() override;
+  cartographer::mapping::MapById<cartographer::mapping::SubmapId, SubmapData>
+  GetAllSubmapData() override;
+  cartographer::transform::Rigid3d GetLocalToGlobalTransform(
+      int trajectory_id) override;
+  cartographer::mapping::MapById<cartographer::mapping::NodeId,
```
Comment (gaschler) [inline cartographer_grpc/mapping/pose_graph_stub.h]: I assume there is no shorter, convenience typedef.

File: cartographer_grpc/mapping/map_builder_stub.cc [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cartographer_grpc/mapping/map_builder_stub.h"
+#include "cartographer_grpc/proto/map_builder_service.pb.h"
```
[the remark this answers is not in the document]
Comment (cschuet) [reply cartographer_grpc/mapping/map_builder_stub.cc]: Done.

File: cartographer_grpc/mapping/pose_graph_stub.cc [hunk 3]
```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2016 The Cartographer Authors
```
[the remark this answers is not in the document]
Comment (cschuet) [reply cartographer_grpc/mapping/pose_graph_stub.cc]: Done.

**Comment by cschuet** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user cschuet. Merge queue now has a length of 1.

**Pull request merged**